NC | CLI | noobaa-cli whitelist command is failing when config.json was never created

### Environment info
- NooBaa Version: 4.19
- Platform: NC

### Actual behavior
1. The following CLI operation will fail if config.json was never created - 
`noobaa-cli whitelist --ips '["127.0.0.1", "192.0.10.0", "3002:0bd6:0000:0000:0000:ee00:0033:6778"]'`

Error - 
```
{
  "error": {
    "code": "WhiteListIPUpdateFailed",
    "message": "Whitelist ip update failed",
    "detail": "/etc/noobaa.conf.d/config.json"
  }
```

### Expected behavior
1. Create config.json if it was never created instead of failing if it does not exist.

### Steps to reproduce
1. run `noobaa-cli whitelist --ips '["127.0.0.1", "192.0.10.0", "3002:0bd6:0000:0000:0000:ee00:0033:6778"]'` while config.json does not exist under /etc/noobaa.conf.d/ .

### More information - Screenshots / Logs / Other output
CLI logs - 
```
% noobaa-cli whitelist --ips '["127.0.0.1", "192.0.10.0", "3002:0bd6:0000:0000:0000:ee00:0033:6778"]'
Error accrued while getting the data from /etc/noobaa.conf.d/config_dir_redirect: Error: ENOENT: no such file or directory, open '/etc/noobaa.conf.d/config_dir_redirect'
config.get_config_root - could not find custom config_root, will use the default config_root  /etc/noobaa.conf.d
load_nsfs_nc_config.setting config.NSFS_NC_CONF_DIR /etc/noobaa.conf.d
config.load_nsfs_nc_config could not find config.json... skipping
(node:29756) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Feb-6 18:26:06.710 [/29756]  [WARN] CONSOLE:: config.load_nsfs_nc_config could not find config.json... skipping
Feb-6 18:26:06.713 [/29756]   [LOG] CONSOLE:: read_rand_seed: reading 32 bytes from /dev/urandom ...
Feb-6 18:26:06.717 [/29756]   [LOG] CONSOLE:: read_rand_seed: got 32 bytes from /dev/urandom, total 32 ...
Feb-6 18:26:06.717 [/29756]   [LOG] CONSOLE:: read_rand_seed: closing fd ...
Feb-6 18:26:06.717 [/29756]   [LOG] CONSOLE:: init_rand_seed: seeding with 32 bytes
Feb-6 18:26:06.718 [/29756]   [LOG] CONSOLE:: init_rand_seed: done
Feb-6 18:26:06.724 [/29756] [ERROR] core.cmd.manage_nsfs:: manage_nsfs.whitelist_ips_management: Error while updation config.json,  path /etc/noobaa.conf.d/config.json Error: Cannot find module '/etc/noobaa.conf.d/config.json'
Require stack:
- noobaa-core/src/cmd/manage_nsfs.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1219:15)
    at Module._load (node:internal/modules/cjs/loader:1045:27)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Module.require (node:internal/modules/cjs/loader:1304:12)
    at require (node:internal/modules/helpers:123:16)
    at whitelist_ips_management (noobaa-core/src/cmd/manage_nsfs.js:700:29)
    at main (noobaa-core/src/cmd/manage_nsfs.js:66:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'noobaa-core/src/cmd/manage_nsfs.js' ]
}
Feb-6 18:26:06.727 [/29756] [EVENT]{"timestamp":"2025-02-06T16:26:06.725Z","host":"hostname1","event":{"code":"noobaa_whitelist_updated_failed","message":"Whitelist updated with IPs failed.","description":"Whitelist updated with IPs failed. Error while updation config.json file with whitelist IPs","entity_type":"NODE","event_type":"INFO","scope":"NODE","severity":"INFO","state":"DEGRADED","pid":29756}}
{
  "error": {
    "code": "WhiteListIPUpdateFailed",
    "message": "Whitelist ip update failed",
    "detail": "/etc/noobaa.conf.d/config.json"
  }
}
```

--- a/src/cmd/manage_nsfs.js
+++ b/src/cmd/manage_nsfs.js
@@ -3,7 +3,6 @@
 
 const dbg = require('../util/debug_module')(__filename);
 const _ = require('lodash');
-const path = require('path');
 const minimist = require('minimist');
 const config = require('../../config');
 const P = require('../util/promise');
@@ -717,6 +716,19 @@ async function set_bucker_owner(bucket_data) {
 ////////////////////
 //// IP WHITELIST //
 ////////////////////
+/**
+ * @returns {Promise<[boolean, object]>} - [config_exists, config_data] - config_exists is a boolean that indicates if the config.json exists
+ * and config_data is the parsed data from the config.json file. returns empty object for config_data if config.json does not exist.
+ */
+async function _get_config_json_if_exist() {
+    try {
+        const config_json = await config_fs.get_config_json();
+        return [true, config_json];
+    } catch (err) {
+        if (err.code !== 'ENOENT') throw err;
+        return [false, {}];
+    }
+}
 
 async function whitelist_ips_management(args) {
     const ips = args.ips;
@@ -724,15 +736,18 @@ async function whitelist_ips_management(args) {
 
     const whitelist_ips = JSON.parse(ips);
     manage_nsfs_validations.validate_whitelist_ips(whitelist_ips);
-    const config_path = path.join(config_fs.config_root, 'config.json');
     try {
-        const config_data = require(config_path);
+        const [config_exists, config_data] = await _get_config_json_if_exist();
         config_data.S3_SERVER_IP_WHITELIST = whitelist_ips;
         const data = JSON.stringify(config_data);
-        await config_fs.update_config_json_file(data);
+        if (config_exists) {
+            await config_fs.update_config_json_file(data);
+        } else {
+            await config_fs.create_config_json_file(data);
+        }
     } catch (err) {
-        dbg.error('manage_nsfs.whitelist_ips_management: Error while updation config.json,  path ' + config_path, err);
-        throw_cli_error(ManageCLIError.WhiteListIPUpdateFailed, config_path);
+        dbg.error('manage_nsfs.whitelist_ips_management: Error while updation config.json, path:', config_fs.config_json_path, err);
+        throw_cli_error(ManageCLIError.WhiteListIPUpdateFailed);
     }
     write_stdout_response(ManageCLIResponse.WhiteListIPUpdated, ips);
 }

--- a/src/test/unit_tests/test_nc_cli.js
+++ b/src/test/unit_tests/test_nc_cli.js
@@ -1055,6 +1055,21 @@ mocha.describe('manage_nsfs cli', function() {
                 assert_error(err, ManageCLIError.InvalidArgument);
             }
         });
+
+        mocha.it('cli add whitelist config file doesnt exist', async function() {
+            //delete existing config file
+            const config_file_path = config_fs.get_config_json_path();
+            await fs_utils.file_delete(config_file_path);
+
+            const ips = ['127.0.0.1']; // IPV4 format
+            const res = await exec_manage_cli(type, '', { config_root, ips: JSON.stringify(ips) });
+            await assert_response('', type, res, ips);
+            const new_config_options = { S3_SERVER_IP_WHITELIST: ips};
+            const config_data = await config_fs.get_config_json();
+            console.log(config_data);
+            assert_whitelist(config_data, new_config_options);
+        });
+
     });
 
 });

